### PR TITLE
Ag/notion source update

### PR DESCRIPTION
**Pull request by dat-a-man**
# Tell us what you do here

- [ ] implementing verified source (please link a relevant issue labelled as `verified source`)
- [ ] fixing a bug (please link a relevant bug report)
- [x] improving, documenting, customizing existing source (please link an issue or describe below)
- [ ] anything else (please link an issue or describe below)

# Relevant issue

issue # 241

# More PR info

This PR addresses the issue where the Notion pipeline expects both id and use_name when fetching data from custom databases. Previously, omitting use_name would lead to a KeyError.

Changes made:

If use_name is not provided, the pipeline will now automatically retrieve it from Notion. For example, if the input is:

```python
selected_database_ids = [{"id": "0517dae9"},{"id":"d8ee2d159a"}]
# The pipeline will fetch the corresponding use_name directly from Notion.
```
This not only prevents the aforementioned error but also improves the user experience, making the pipeline a bit more user-friendly and intuitive.

Users can also give input as:

```python
selected_database_ids = [{"id": "0517dae9", "use_name":"name1"},{"id":"d8ee2d159a"}]
# the pipeline takes the first use_name from the input provided and fetches the second one automatically. 
```

Please review and let me know if there are any additional changes needed.

### CODE REVIEW

File: sources/notion/helpers/client.py [hunk 1]
```diff
@@ -151,3 +151,14 @@ def search(
             next_cursor = response.get("next_cursor")
             has_more = next_cursor is not None
             start_cursor = next_cursor
+
+    def get_database(self, database_id: str) -> Dict[str, Any]:
```
Comment (AstrakhantsevaAA) [inline sources/notion/helpers/client.py:155]: linter failed. change Dict[str, Any] to Any.

Comment (AstrakhantsevaAA) [reply sources/notion/helpers/client.py:155]: the rest looks good, well done))

Comment (AstrakhantsevaAA) [reply sources/notion/helpers/client.py:155]: don't forget to edit docstring

Comment (dat-a-man) [reply sources/notion/helpers/client.py]: Changes done. Please see.

Comment (dat-a-man) [reply sources/notion/helpers/client.py]: PR passed the checks @AstrakhantsevaAA 👍  please review.  Thanks

**Review by AstrakhantsevaAA — APPROVED** [review]

**Pull request merged**